src/term/term_to_net.rs

```diff
@@ -100,6 +109,7 @@ fn encode_term(
   vars: &mut Vec<(Port, Option<Port>)>,
   global_vars: &mut HashMap<Name, (Port, Port)>,
   dups: &mut u8,
```

Comment:
Could you join `dups` and `tag_storage` into a single "dup label generator" object?
Then you could have something like `LabelGenerator::generate(&mut self, tag: &Option<Name>)` and one argument fewer here.

This function is growing a lot in number of state variables and it becomes a bit hard to keep track of.